### PR TITLE
[prometheus-redis-exporter] support multiple targets

### LINKED ISSUE #2198: [prometheus-redis-exporter] support multiple targets
### Is your feature request related to a problem ?

Now redis-exporter has supported multiple target, through `/scrape` https://github.com/oliver006/redis_exporter#prometheus-configuration-to-scrape-multiple-redis-hosts 

```
scrape_configs:
  ## config for the multiple Redis targets that the exporter will scrape
  - job_name: 'redis_exporter_targets'
    static_configs:
      - targets:
        - redis://first-redis-host:6379
        - redis://second-redis-host:6379
        - redis://second-redis-host:6380
        - redis://second-redis-host:6381
    metrics_path: /scrape
    relabel_configs:
      - source_labels: [__address__]
        target_label: __param_target
      - source_labels: [__param_target]
        target_label: instance
      - target_label: __address__
        replacement: <<REDIS-EXPORTER-HOSTNAME>>:9121

  ## config for scraping the exporter itself
  - job_name: 'redis_exporter'
    static_configs:
      - targets:
        - <<REDIS-EXPORTER-HOSTNAME>>:9121
```
This could be useful when we have many redis to scrape and only want one single exporter

### Describe the solution you'd like.

We can render multiple service monitor like blackbox exporter did in the chart https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-blackbox-exporter/templates/servicemonitor.yaml 

### Describe alternatives you've considered.

NONE

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by LeoQuote**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #2198 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
You have multiple breaking changes in this pr:
* default labels change -> whats the upgrade path?
* customLabels were removed -> why?
* service port variable changed -> why?


You also have to sign your commits.

**Comment by LeoQuote** [comment]
@monotek will do, thanks for the review

**Comment by LeoQuote** [comment]
@monotek 

default labels was changed, but `managed-by: Helm` label is still present, helm upgrade can handle the upgrade, I added a note in readme to notify people that selectorLabels are changed.

I also added detailed comments in values.yaml to clearify where customLabels and labels will be used.

service port variable is changed to service port name , I think this could be more Intuitive, targetPort is actually container port, sometimes this could be misleading.

**Review by monotek — CHANGES_REQUESTED** [review]
Upgrade from 4.8.0 fails. Please provide upgradepath.

```
W0629 10:27:05.881840  124326 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0629 10:27:05.882859  124326 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
W0629 10:27:05.887939  124326 warnings.go:70] policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
Error: UPGRADE FAILED: cannot patch "prom-redis-prometheus-redis-exporter" with kind Deployment: Deployment.apps "prom-redis-prometheus-redis-exporter" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/instance":"prom-redis", "app.kubernetes.io/name":"prometheus-redis-exporter"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable

```

File: charts/prometheus-redis-exporter/README.md
Comment (monotek) [inline charts/prometheus-redis-exporter/README.md:65]: You could keep your helm history if you delete the deployment before upgrading:

```suggestion
kubectl delete deployment -l app=prometheus-redis-exporter
helm upgrade -i prometheus-redis-exporter prometheus-community/prometheus-redis-exporter
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-redis-exporter/README.md
Comment (monotek) [inline charts/prometheus-redis-exporter/README.md:55]:
```suggestion
From 5.0.0 redis exporter is using the [Kubernetes recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/). Therfore youl have to delete the deployment before your upgrade.
```

File: charts/prometheus-redis-exporter/README.md
Comment (monotek) [inline charts/prometheus-redis-exporter/README.md:56]: Line 56 - 62 can be deleted.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-redis-exporter/README.md
Comment (LeoQuote) [reply charts/prometheus-redis-exporter/README.md:66]: Thank you! I didn't know that could be done

File: charts/prometheus-redis-exporter/README.md
Comment (LeoQuote) [reply charts/prometheus-redis-exporter/README.md]: It is deleted now.

File: charts/prometheus-redis-exporter/README.md
Comment (monotek) [inline charts/prometheus-redis-exporter/README.md:55]: Sorry, some typos...

```suggestion
From 5.0.0 redis exporter is using the [Kubernetes recommended labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/). Therefore you have to delete the deployment before you upgrade.
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by LeoQuote** [comment]
markdown lint found serveral issues, but these seems not to be real issue , 

`helm repo` and `github` are very typical usage , especially in this repo

**Review by monotek — APPROVED** [review]

**Comment by LeoQuote** [comment]
Thanks for your edit ! @monotek I think this chart could work now

**Comment by monotek** [comment]
Let's wait for the reviews of @acondrat or @zanhsieh as they are codeowners.

**Review by zanhsieh — APPROVED** [review]

**Comment by Sameer1808** [comment]
@LeoQuote the multiple targets thing still isn't working for me, here is the config for the exporter :

# If serviceMonitor.multipleTarget is enabled, this configuration is actually not used
#redisAddress: redis://<redis-server-1>:6379


serviceMonitor:
  # When set true then use a ServiceMonitor to configure scraping
  enabled: true
  multipleTarget: true
  telemetryPath: /scrape
  targets:
  # for every targets, url and name must be set,
  # an individual additionalRelabeling can be set for every target
  - url: redis://<redis-server-1>:6379
    name: redis-server-1
  - url: redis://<redis-server-2>:6379
    name: redis-server-2
  additionalMetricsRelabels: {}
  additionalRelabeling: []


we have tried with and without passing the telemetry path but still there are no metrics in prometheus, we have checked the connectivity..
its working if we pass single host in `redisAddress` .. let us know if we are missing anything here

**Comment by LeoQuote** [comment]
@Sameer1808 hi, would your open a new issue and refer to this pull request?
more info is required to debug this problem , like the redis exporter image you’re using, full values file if convenient.

**Comment by shubhamsetia05** [comment]
@LeoQuote as per your suggestion, a new issue has been opened - https://github.com/prometheus-community/helm-charts/pull/2199, also attached the values file we are using.

**Pull request merged**
